src/App.tsx

```diff
@@ -1,61 +1,146 @@
+import { useState } from 'react';
+
 import './App.scss';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const modifiedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(modifiedTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+  const [titleErrorValidation, setTitleErrorValidation] = useState(false);
+
+  const addNewTodo = () => {
+    const uniqueId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+    const newTodo = {
+      id: uniqueId,
+      title,
+      userId,
+      completed: false,
+      user: getUserById(userId),
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const clearForm = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const titleHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+    setTitleError(false);
+  };
+
+  const userHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
```suggestion
  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(e.target.value);
    setTitleError(false);
  };

  const handleUserChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
Thank you for the link, I renamed them